gen/build_deploy/bash.py

```diff
@@ -27,19 +27,23 @@
 from pkgpanda.util import logger
 
 
-def calculate_fault_domain_detect_contents(fault_domain_detect_filename, fault_domain_enabled):
-    if fault_domain_enabled == 'false':
-        return ''
-    return yaml.dump(open(fault_domain_detect_filename, encoding='utf-8').read())
+def calculate_fault_domain_detect_contents(fault_domain_detect_filename):
+    if os.path.exists(fault_domain_detect_filename):
+        return yaml.dump(open(fault_domain_detect_filename, encoding='utf-8').read())
+    return ''
+
 
+def calculate_fault_domain_enabled(fault_domain_detect_filename, fault_domain_detect_contents):
+    if fault_domain_detect_contents:
+        return 'true'
 
-def calculate_fault_domain_enabled(fault_domain_detect_filename):
     try:
         with open(fault_domain_detect_filename):
-            pass
+            return 'true'
     except FileNotFoundError:
-        return 'false'
-    return 'true'
+        pass
+
+    return 'false'
```

Comment:
It looks like this stanza is doing the same thing as before: in both versions if the file exists it returns `'true'`, if the file doesn't exist it returns `'false'`.

And actually I think we can get rid of this stanza entirely (except for the `return 'false'` at the end). Now we're conditionally setting `fault_domain_detect_contents` based on the presence of the file, and then setting `fault_domain_enabled` based on that, so this is duplicating the file existence check that `calculate_fault_domain_detect_contents()` already does.